### DISCUSSION TITLE
Calling an rpc functions with uuid parameters with a null value using GET is causing an error

### DISCUSSION THREAD

**Discussion by christiaanwesterbeek** [PostgREST/postgrest#2533] [Q&A] [answered]
### Environment

* PostgreSQL version: 13
* PostgREST version: 10

### Description of issue

Calling an rpc functions with uuid parameters with a null value using GET is causing the following error

> Error code 22P02: invalid input syntax for type uuid: "null"

The request is like this:
GET /rpc/my-function?x_id=12345678-1234-1234-1234-123456789012&y_id=null

I would like to call the function using GET, since it's stable but in some cases the y_id value is null and the function requires it to be passed. I know there's a special case built in for calling a function with GET and passing an array literal, as in {1,2,3,4}. So I hope I was able to pass a `null` value instead of the sql statement becoming 'null'...

**Comment by wolfgangwalther** [comment]
This is expected behavior.

You can add a `DEFAULT NULL` clause to the `y_id` argument. Then you can call it with both `x_id` only or `x_id` and `y_id` together.

**Comment by christiaanwesterbeek** [comment, marked as answer]
I've tried that but unfortunately I get because after y_id comes a few other parameters that cannot have a default (I think).

> ERROR:  input parameters after one with a default value must also have defaults

It's a bit too much work to change the function signature, so I'll revert to the following workaround.

That's to instead of calling y_id with `null`, I'm calling it with `00000000-0000-0000-0000-000000000000`.

GET /rpc/my-function?x_id=12345678-1234-1234-1234-123456789012&y_id=00000000-0000-0000-0000-000000000000

And in my function I'm checking it like this:

`case when "my-function".y_id = extensions.uuid_nil() then ... end`

**Reply by wolfgangwalther** [reply]
> I've tried that but unfortunately I get because after y_id comes a few other parameters that cannot have a default (I think).

If the `DEFAULT` solution is not going to work for you, you can go the overloaded way:

```sql
CREATE FUNCTION yourfunc(x_id UUID, y_id UUID, other_parameters) ...;

CREATE FUNCTION yourfunc(x_id UUID, other_parameters) RETURNS <the same as yourfunc>
LANGUAGE SQL AS $$
  SELECT yourfunc(x_id, NULL, other_parameters);
$$;
```

This will allow you to call `/rpc/yourfunc` both with and without `y_id` - without changing the function you already have.
